### PR TITLE
feat: [SRM-9913]: 3rd part API call log and execution log for monito…

### PR DESCRIPTION
**Pull request by vasuki-kalluri**
…red Service details page

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Review by shashidhargg — APPROVED** [review]

**Comment by vasuki-kalluri** [comment]
retrigger coverage

**Comment by vasuki-kalluri** [comment]
retrigger jest

**Comment by vasuki-kalluri** [comment]
retrigger coverage

**Comment by vasuki-kalluri** [comment]
retrigger coverage

File: src/modules/85-cv/pages/monitored-service/components/ServiceHealth/components/MetricsAndLogs/__tests__/MetricsAndLogs.test.tsx [hunk 1]
```diff
@@ -64,4 +71,31 @@ describe('Unit tests for MetricsAndLogs', () => {
     const { getByTestId } = render(<WrapperComponent {...props} />)
     expect(getByTestId('analysis-image-view')).toBeTruthy()
   })
+
+  test('should open the LogContent modal and render MonitoredServiceLog with typ ApiCallLog by clicking the Execution Logs button', async () => {
```
Comment (shashidhargg) [inline src/modules/85-cv/pages/monitored-service/components/ServiceHealth/components/MetricsAndLogs/__tests__/MetricsAndLogs.test.tsx:75]: Please change the test title, here we are clicking the External API Calls link.

File: src/modules/85-cv/hooks/useLogContentHook/views/LogContentHeader/LogContentHeader.tsx [hunk 2]
```diff
@@ -89,10 +118,20 @@ const LogContentHeader: React.FC<LogContentHeaderProps> = ({
...
+          ) : isMonitoredDetailsStep ? (
+            <Select
+              value={{
+                label: `${getString('pipeline.verification.healthSourceLabel')}: ${healthSource?.label}`,
+                value: healthSource?.value ?? ''
+              }}
+              items={getHealthSourceOptions(getString, monitoredService?.resource)}
+              onChange={handleHealthSource}
+              disabled={monitoredServiceLoading}
+            />
```
Comment (shashidhargg) [inline src/modules/85-cv/hooks/useLogContentHook/views/LogContentHeader/LogContentHeader.tsx:134]: Please have a single Select for both. Just use diff data.
```
{(isVerifyStep || isMonitoredDetailsStep ) ? (
  <Select
    value={{
      label: `${getString('pipeline.verification.healthSourceLabel')}: ${healthSource?.label}`,
      value: healthSource?.value ?? ''
    }}
    items={getHealthSourceOptions(getString, isVerifyStep ? verifyStep?.resource : monitoredService?.resource)}
    onChange={handleHealthSource}
    disabled={loading || monitoredServiceLoading}
  />
```

File: src/modules/85-cv/hooks/useLogContentHook/views/LogContentHeader/LogContentHeader.tsx [hunk 3]
```diff
@@ -22,16 +22,38 @@ const LogContentHeader: React.FC<LogContentHeaderProps> = ({
...
+    data: monitoredService,
+    loading: monitoredServiceLoading,
+    refetch: refetchMonitoredHealthScore
+  } = useGetAllHealthSourcesForServiceAndEnvironment({
+    queryParams: {
+      orgIdentifier,
+      projectIdentifier,
+      accountId,
+      serviceIdentifier: serviceName!,
+      environmentIdentifier: envName!
```
Comment (shashidhargg) [inline src/modules/85-cv/hooks/useLogContentHook/views/LogContentHeader/LogContentHeader.tsx:48]: Please use empty strings as default values for `serviceName` and `envName` at the level of the props and remove the force type.

**Review by shashidhargg — CHANGES_REQUESTED** [review]
Please check the comments.

File: src/modules/85-cv/hooks/useLogContentHook/__test__/VerifyStepLogContentWithMockActions.test.tsx [hunk 4]
```diff
@@ -37,6 +37,9 @@ jest.mock('services/cv', () => ({
   useGetVerifyStepLogs: jest
     .fn()
     .mockImplementation(() => ({ data: executionLogsResponse, loading: false, error: null, refetch: jest.fn() })),
+  useGetAllHealthSourcesForServiceAndEnvironment: jest
+    .fn()
+    .mockImplementation(() => ({ data: executionLogsResponse, loading: false, error: null, refetch: jest.fn() })),
```
[the remark this answers is not in the document]
Comment (vasuki-kalluri) [reply src/modules/85-cv/hooks/useLogContentHook/__test__/VerifyStepLogContentWithMockActions.test.tsx:42]: Done

File: src/modules/85-cv/hooks/useLogContentHook/views/LogContentHeader/LogContentHeader.tsx [hunk 5]
```diff
@@ -22,16 +22,38 @@ const LogContentHeader: React.FC<LogContentHeaderProps> = ({
...
+  const isMonitoredDetailsStep = Boolean(monitoredServiceIdentifier)
+
+  const { orgIdentifier, projectIdentifier, accountId } = useParams<ProjectPathProps & { identifier: string }>()
 
-  const { data, loading, refetch } = useGetVerifyStepHealthSources({
+  const {
+    data: monitoredService,
+    loading: monitoredServiceLoading,
+    refetch: refetchMonitoredHealthScore
+  } = useGetAllHealthSourcesForServiceAndEnvironment({
```
[the remark this answers is not in the document]
Comment (vasuki-kalluri) [reply src/modules/85-cv/hooks/useLogContentHook/views/LogContentHeader/LogContentHeader.tsx:42]: Done

File: src/modules/85-cv/hooks/useLogContentHook/views/MonitoredServiceLogContent.tsx [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+        queryParams: {
+          accountId,
+          orgIdentifier,
+          projectIdentifier,
+          logType,
+          errorLogsOnly,
+          startTime,
+          endTime,
+          pageSize: totalItems
+        }
```
[the remark this answers is not in the document]
Comment (vasuki-kalluri) [reply src/modules/85-cv/hooks/useLogContentHook/views/MonitoredServiceLogContent.tsx:72]: Done

File: src/modules/85-cv/pages/monitored-service/components/ServiceHealth/components/MetricsAndLogs/__tests__/MetricsAndLogs.test.tsx (hunk 1 above)
Comment (vasuki-kalluri) [reply src/modules/85-cv/pages/monitored-service/components/ServiceHealth/components/MetricsAndLogs/__tests__/MetricsAndLogs.test.tsx:75]: Done

File: src/modules/85-cv/hooks/useLogContentHook/views/LogContentHeader/LogContentHeader.tsx (hunk 2 above)
Comment (vasuki-kalluri) [reply src/modules/85-cv/hooks/useLogContentHook/views/LogContentHeader/LogContentHeader.tsx:134]: Done

File: src/modules/85-cv/hooks/useLogContentHook/views/LogContentHeader/LogContentHeader.tsx (hunk 3 above)
Comment (vasuki-kalluri) [reply src/modules/85-cv/hooks/useLogContentHook/views/LogContentHeader/LogContentHeader.tsx:48]: Done

File: src/modules/85-cv/hooks/useLogContentHook/useLogContentHook.tsx [hunk 7]
```diff
@@ -13,13 +13,17 @@ import { Dialog } from '@harness/uicore'
...
+import MonitoredServiceLogContent from './views/MonitoredServiceLogContent'
 import css from './useLogContentHook.module.scss'
 
 export const useLogContentHook = ({
   verifyStepExecutionId,
   sloIdentifier,
-  serviceName,
-  envName
+  serviceName = '',
+  envName = '',
```
[the remark this answers is not in the document]
Comment (vasuki-kalluri) [reply src/modules/85-cv/hooks/useLogContentHook/useLogContentHook.tsx]: Done

**Comment by vasuki-kalluri** [comment]
retrigger coverage

**Review by shashidhargg — APPROVED** [review]

**Review by diptiman-adak — DISMISSED** [review]

File: src/modules/85-cv/hooks/useLogContentHook/views/MonitoredServiceLogContent.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (shashidhargg) [reply src/modules/85-cv/hooks/useLogContentHook/views/MonitoredServiceLogContent.tsx:72]: Why did we remove it again?

**Review by shashidhargg — APPROVED** [review]

**Review by diptiman-adak — APPROVED** [review]
Approving since no other reviewer is available that time.

**Comment by vasuki-kalluri** [comment]
retrigger cypress

**Comment by vasuki-kalluri** [comment]
trigger cypress

**Review by shashidhargg — APPROVED** [review]

**Review by tan-nhu — APPROVED** [review]

**Pull request merged**